src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -851,21 +850,26 @@ public function delete($id, $checkSecurity = false)
      */
     public function move($id, $locale, $destCollection)
     {
-        try {
-            $mediaEntity = $this->mediaRepository->findMediaById($id);
-
-            if ($mediaEntity === null) {
-                throw new MediaNotFoundException($id);
-            }
+        $mediaEntity = $this->mediaRepository->findMediaById($id);
 
-            $mediaEntity->setCollection($this->em->getReference(self::ENTITY_NAME_COLLECTION, $destCollection));
+        if ($mediaEntity === null) {
+            throw new MediaNotFoundException($id);
+        }
 
-            $this->em->flush();
+        // use find rather than ->getReference as its behavior when a collection doesn't
+        // exist is universal (returns NULL) and using a proxy provides no worth-while
+        // performance benefit for move.
```

Comment:
Hmm, I guess the issue is actually that, when `flush()` was being called, the foreign key constraint was failing and throwing the DBALException. As Sqlite had no foreign key support, this was failing.

I will revert this as I think we are working to a policy of not working-around Sqlite foreign key support.